CIP-0072/version_1_offchain.json

```diff
@@ -11,63 +16,92 @@
+        },
+        "description": "URLs to screenshots of DApp actual pages. We recommend to share screenshots from the dApp usage itself."
+      },
       "social": {
         "type":"object",
         "properties": {
-          "website": {
-            "type":"string",
-            "description": "dApps website link.",
-            "pattern": "(https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|www\\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9]+\\.[^\\s]{2,}|www\\.[a-zA-Z0-9]+\\.[^\\s]{2,})"
+          "companyName": {
+            "type": "string",
+            "description": "Name of the company developing/maintaining the dApp",
+            "maxLength": 50
+          },
+          "companyEmail": {
+            "type": "string",
+            "description": "Email of the company developing/maintaining the dApp",
+            "maxLength": 150,
+            "format": "email"
+          },
+          "companyWebsite": {
+            "type": "string",
+            "description": "Website of the company developing/maintaining the dApp",
+            "maxLength": 200,
+            "pattern": "((https?|ftp|ipfs|ipns):\/\/[\\u00C0-\\u017F-a-zA-Z0-9])"
           },
-          "twitter": {
+          "X.com": {
             "type":"string",
-            "description": "An optional Twitter link.",
-            "pattern": "(https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|www\\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9]+\\.[^\\s]{2,}|www\\.[a-zA-Z0-9]+\\.[^\\s]{2,})"
+            "description": "An optional X.com link.",
+            "maxLength": 200,
+            "pattern": "((https?|ftp|ipfs|ipns):\/\/[\\u00C0-\\u017F-a-zA-Z0-9])"
           },
           "github": {
             "type":"string",
             "description": "An optional Github link.",
-            "pattern": "(https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|www\\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9]+\\.[^\\s]{2,}|www\\.[a-zA-Z0-9]+\\.[^\\s]{2,})"
+            "maxLength": 200,
+            "pattern": "((https?|ftp|ipfs|ipns):\/\/[\\u00C0-\\u017F-a-zA-Z0-9])"
           },
           "discord": {
             "type":"string",
             "description": "An optional Discord link.",
-            "pattern": "(https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|www\\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\\.[^\\s]{2,}|https?:\/\/(?:www\\.|(?!www))[a-zA-Z0-9]+\\.[^\\s]{2,}|www\\.[a-zA-Z0-9]+\\.[^\\s]{2,})"
+            "maxLength": 200,
+            "pattern": "((https?|ftp|ipfs|ipns):\/\/[\\u00C0-\\u017F-a-zA-Z0-9])"
           }
         },
-        "required": ["website"]
+        "required": ["companyWebsite"]
       },
```

Comment:
Would it not be better to define this as an array of objects that might include a string identifier (i.e. X.com, GitHub, Twitter, Discord, etc) so that we do not need to revisit the spec any time a potentially new social media platform comes online that services may want to use to promote and interact with their customer base?

Reply:
Proposed possible structure for greater flexibility in the future:
```json
{
  "social": {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "name": {
          "type": "string",
          "description": "The platform or resource identifier (GitHub, Website, X.com, etc)"
        },
        "link": {
          "type": "string",
          "pattern": "((https?|ftp|ipfs|ipns):\/\/[\\u00C0-\\u017F-a-zA-Z0-9])"
        }
      }
  }
}
```

Reply:
yes @Crypto2099 I think making this a list of key-value pairs is a great way to future proof it, but also recall from CIP meeting discussions at the time (https://github.com/cardano-foundation/CIPs/pull/355#pullrequestreview-1428139625) that the working group for this spec was trying to get an MVP proposal off the ground.  (There is a chance that I have this mixed up with a different CIP so perhaps @matiwinnetou can confirm or deny this.)

What I recall from those objectives is that the first active versions of this standard were intended to be demonstrated before pressing for evolution.  If that's still a goal, then putting the social links in key-value pairs would be better scheduled for a future version of the schema.  cc @marcin-mazurek @danielmain

Reply:
Thank you @Crypto2099 I added your suggestion!

Reply:
Thanks @Crypto2099 for proposing this & @danielmain for accepting it at this formative stage 🤓 ... resolving since changes have been committed.

Reply:
> (There is a chance that I have this mixed up with a different CIP so perhaps @matiwinnetou can confirm or deny this.)

p.s. the CIP I recalled for which having an earlier (perhaps imperfect) version release would have been desirable was the related CIP-0096 as per the discussion here: https://github.com/cardano-foundation/CIPs/pull/499